### PR TITLE
fix devices sprites on layer 1 and 5

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
This atomized PR fixes the sprites for layer 1 and 5 of the following devices:
-scrubber
-unary vent
-passive vent
-portable connector
_-filter_ filters don't work well on layer 1 and 5, so they get the axe
_-mixer_ mixers don't work well on layer 1 and 5, so they get the axe
The how:
instead of just moving the node, we'll move the entire thing because sprites bad and can't fit

- [x] PR ready

## Why It's Good For The Game
better sprites

## Changelog
:cl:
fix: fix sprite jank on layer 1 and 5
/:cl:

### CODE REVIEW

**Comment by Ghilker** [comment]
f wait i need to fix the filters and mixers

**Comment by Ghilker** [comment]
filters and mixers don't work well on layer 1 and 5

**Comment by Ghilker** [comment]
added 
aaa

**Comment by Ghilker** [comment]
f

**Comment by trerri** [comment]
If anyone ever considers merging this please realize that it implies that EVERY roundstart vent and scrubber will be misaligned to the tiles it's on, since distro and waste are not on layer 3.
image

Also, killing the layers of compatibility for trinary components kinda blows. Layer 1 and 5 mixers/filters are very useful.

The first image with the offsets on the components is cool though (because it was my idea 😎)

**Comment by Ghilker** [comment]
> If anyone ever considers merging this please realize that it implies that EVERY roundstart vent and scrubber will be misaligned to the tiles it's on, since distro and waste are not on layer 3.
> image
> 
> Also, killing the layers of compatibility for trinary components kinda blows. Layer 1 and 5 mixers/filters are very useful.
> 
> The first image with the offsets on the components is cool though (because it was my idea 😎)

i know the problem of misaligned vents and scrubbers, but that's the price for changing all distro and waste to 1 and 3 (now 2 and 4) 

the filters and mixers for 1 and 5 is still a maybe, if you can post picture of them at layer 1 and 5 people and maintainers could think about them and if they are no big deal i'll keep them

**Comment by trerri** [comment]
This implies EVERY vent or scrubber at roundstart will be misaligned like this 
image
This also kills a big part of extra layers which was mixers and filters.

The first image with the offsets is really cool though (because it was my idea 😎)

**Comment by trerri** [comment]
> the filters and mixers for 1 and 5 is still a maybe, if you can post picture of them at layer 1 and 5 people and maintainers could think about them and if they are no big deal i'll keep them
image
shes a beaut

**Comment by Ghilker** [comment]
Layer 2 and 4 are just slightly misaligned and they don't go over the tile

I agree that filter and mixers are useful but if there isn't a way to make them work ill axe them

**Comment by trerri** [comment]
Going off this image: 
image
You could pixel shift it to the north east a little bit, and when it's in layer 2 or 4 instead of 1 or 5 you could pixel shift it still, but only half of the amount of the edge layers, for consistency's sake. Here's a shitty mockup of how that would look:
image
And for layers 2 and 4:
image

trerri singlehandedly saves tgstation atmospherics once again, what a hero

**Comment by Ghilker** [comment]
scrubbers and vents now align back in the center of the tile if covered by floor

**Comment by trerri** [comment]
i love ghilker

**Comment by LemonInTheDark** [comment]
We love ghilker

**Review by Twaticus — APPROVED** [review]

File: code/modules/atmospherics/machinery/components/binary_devices/binary_devices.dm [hunk 1]
```diff
@@ -15,3 +15,10 @@
 
 /obj/machinery/atmospherics/components/binary/getNodeConnects()
 	return list(turn(dir, 180), dir)
+
+///Used by binary devices to set what the offset will be for each layer
+/obj/machinery/atmospherics/components/binary/proc/set_overlay_offset(var/pipe_layer)
+	if(pipe_layer == 1 || pipe_layer == 3 || pipe_layer == 5)
+		return 1
+	if(pipe_layer == 2 || pipe_layer == 4)
+		return 2
```
Comment (Rohesie) [inline code/modules/atmospherics/machinery/components/binary_devices/binary_devices.dm:24]:
```suggestion
	switch(pipe_layer)
		if(1, 3, 5)
			return 1
		if(2, 4)
			return 2
```

File: code/modules/atmospherics/machinery/components/trinary_devices/trinary_devices.dm [hunk 2]
```diff
@@ -46,3 +46,11 @@ Housekeeping and pipe network stuff
...
 
 	return list(node1_connect, node2_connect, node3_connect)
+
+/obj/machinery/atmospherics/components/trinary/proc/set_overlay_offset(var/pipe_layer)
+	if(pipe_layer == 1)
+		return 1
+	else if(pipe_layer == 5)
+		return 5
+	else
+		return 0
```
Comment (Rohesie) [inline code/modules/atmospherics/machinery/components/trinary_devices/trinary_devices.dm:56]:
```suggestion
	switch(pipe_layer)
		if(1)
			return 1
		if(5)
			return 5
		else
			return 0
```

**Pull request merged**